scripts/validate_docstrings.py

```diff
@@ -331,6 +336,33 @@ def parameter_mismatches(self):
 
         return errs
 
+    @contextlib.contextmanager
+    def _write_examples_code_to_temp_file(self):
+        """
+        Generate file with source code from examples section.
+        """
+        content = ''.join(('import numpy as np; '
+                           'import pandas as pd  # noqa: F401,E702\n',
+                           *self.examples_source_code))
+        with tempfile.NamedTemporaryFile(mode='w') as file:
+            file.write(content)
+            file.flush()
+            yield file
+
+    def validate_pep8(self):
+        if self.examples:
```

Comment:
small thing, but my preference is having:
```
if not self.examples:
    return
```
this avoids having the rest of the function indented, and in my opinion increases readability.